tests/unit/bokeh/command/subcommands/test_serve.py

```diff
@@ -471,6 +482,50 @@ def check_error(args):
         pytest.fail(f"command {cmd} unexpected successful")
     return out
 
+def test_unix_socket_with_port() -> None:
+    unix_socket = "test.sock"
+    out = check_error(["--unix-socket", unix_socket, "--port", "5000"]).strip()
+    expected = "--port arg is not supported with a unix socket"
+    assert expected == out
+
+def test_unix_socket_with_invalid_args() -> None:
+    invalid_args = ['address', 'allow-websocket-origin', 'ssl-certfile', 'ssl-keyfile']
+    for arg in invalid_args:
+        unix_socket = "test.sock"
+        out = check_error(["--unix-socket", unix_socket, f"--{arg}", "value"]).strip()
+        expected = "['address', 'allow_websocket_origin', 'ssl_certfile', 'ssl_keyfile', 'port'] args are not supported with a unix socket"
+        assert expected == out
+
+@flaky(max_runs=10)
+def test_unix_socket() -> None:
+    # Unix sockets can not be bind to a file
+    # via python in windows currently
+    # Reference: https://github.com/python/cpython/issues/77589
+    if "win" in sys.platform:
```

Comment:
@govinda18 Yes the test should be skipped on windows, but we should should use pytest APIs to do that so that the test is properly reported as a skipped rather than passed on windows. Something like:

```
@pytest.mark.skipif(sys.platform == "win32", "unix sockets not available on windows")
def test_unix_socket() -> None:
```

Additionally this reminds me that the option help and docs should explicitly mention that that option does not work on windows, and if in case it is set on windows, a `RuntimeError` with a message to that effect should be raised. 

Thanks for your perserverance @govinda18 any PR with a UX and/or platform-specific component will always end up having a lot of small details, but I think this PR is close

Reply:
Thanks, @bryevdv. This is done. Currently, a single test fails, I am not really sure what went wrong there. Can you check once and let me know if there is something wrong at my end?